### PR TITLE
Add suspended status

### PR DESCRIPTION
**Pull request by dthomson25**

### CODE REVIEW

**Comment by jessesuen** [comment]
I feel like there are other places where we may switch for health. For example prometheus metrics need to include Suspended in the gauge. Also, need to double check that sync hooks are not affected by this change.

**Comment by jessesuen** [comment]
metrics.go
```golang
	healthStatus := app.Status.Health.Status
	addGauge(descAppHealthStatus, boolFloat64(healthStatus == argoappv1.HealthStatusUnknown || healthStatus == ""), argoappv1.HealthStatusUnknown)
	addGauge(descAppHealthStatus, boolFloat64(healthStatus == argoappv1.HealthStatusProgressing), argoappv1.HealthStatusProgressing)
	addGauge(descAppHealthStatus, boolFloat64(healthStatus == argoappv1.HealthStatusHealthy), argoappv1.HealthStatusHealthy)
	addGauge(descAppHealthStatus, boolFloat64(healthStatus == argoappv1.HealthStatusDegraded), argoappv1.HealthStatusDegraded)
	addGauge(descAppHealthStatus, boolFloat64(healthStatus == argoappv1.HealthStatusMissing), argoappv1.HealthStatusMissing)
```

**Comment by dthomson25** [comment]
I added the new suspended status to the metrics.go and a test for the deployment.

File: cmd/argocd/commands/app.go [hunk 1]
```diff
@@ -934,19 +935,22 @@ func NewApplicationWaitCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
...
 			appName := args[0]
 			acdClient := argocdclient.NewClientOrDie(clientOpts)
-			_, err := waitOnApplicationStatus(acdClient, appName, timeout, watchSync, watchHealth, watchOperations, nil)
+			_, err := waitOnApplicationStatus(acdClient, appName, timeout, watchSync, watchHealth, watchOperations, watchSuspended, nil)
 			errors.CheckError(err)
 		},
 	}
 	command.Flags().BoolVar(&watchSync, "sync", false, "Wait for sync")
 	command.Flags().BoolVar(&watchHealth, "health", false, "Wait for health")
+	command.Flags().BoolVar(&watchSuspended, "suspended", false, "Wait for suspended")
```
Comment (jessesuen) [inline cmd/argocd/commands/app.go]: It seems to me that health and suspended are mutually exclusive flags (e.g an application will never be both healthy and suspended). If that is correct, we should error when they supply both.

**Review by alexmt — APPROVED** [review]
LGTM besides @jessesuen 's comment about mutually exclusive flags.

**Comment by dthomson25** [comment]
Fixed the comment about the mutually exclusive flags

**Pull request merged**